FEXHeaderUtils/FEXHeaderUtils/Filesystem.h

```diff
@@ -14,4 +22,270 @@ namespace FHU::Filesystem {
   inline bool Exists(const char *Path) {
     return access(Path, F_OK) == 0;
   }
+
+  inline bool Exists(const fextl::string &Path) {
+    return access(Path.c_str(), F_OK) == 0;
+  }
+
+  /**
+   * @brief Creates a directory at the provided path.
+   *
+   * @param Path The path to create a directory at.
+   *
+   * @return True if the directory was created or already exists.
```

Comment:
This is not consistent with `std::create_directory`, which returns false if the directory already existed. (This is different from `std::create_directories`, so CreateDirectories doesn't need changes in behavior but needs to revisit how it uses CreateDirectory internally)

Reply:
Resolved by returning an enum result about what happened rather than boolean.